### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by alexdeem**
Full specification support; but only a small subset of formatting commands
Strict nesting is not validated, and there are some obvious performance and cleanliness improvements to be made.

### CODE REVIEW

File: Source/EnrichedTextStyle.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+}
+
+public struct EnrichedTextStyle {
+    public var options: EnrichedTextStyleOptions
+}
+
+extension EnrichedTextStyle {
+    internal mutating func apply(command:Substring, param:Substring?) {
+        switch (command.lowercased()) {
```
Comment (alexdeem) [inline Source/EnrichedTextStyle.swift:23]: `lowercased()` = opportunity for performance improvement

File: Source/EnrichedTextStyle.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            options.insert(.italic)
+        case "underline":
+            options.insert(.underline)
+        default:
+            break
+        }
+    }
+
+    internal mutating func negate(command:Substring) {
+        switch (command.lowercased()) {
```
Comment (alexdeem) [inline Source/EnrichedTextStyle.swift:36]: `lowercased()` = opportunity for performance improvement

File: Source/Internal/Scanner.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        if (endIndex == unicodeScalars.endIndex) {
+            throw EnrichedText.Error.malformed(position: endIndex, reason: "Unterminated Command")
+        }
+        if (unicodeScalars[endIndex] != ">") {
+            throw EnrichedText.Error.malformed(position: endIndex, reason: "Unexpected Character in Command")
+        }
+        currentIndex = unicodeScalars.index(after: endIndex)
+
+        let command = string[startIndex..<endIndex]
+        if (command.caseInsensitiveCompare("nofill") == .orderedSame) {
```
Comment (alexdeem) [inline Source/Internal/Scanner.swift:77]: `caseInsensitiveCompare` = opportunity for performance improvement

File: Source/Internal/Scanner.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        if (negation) {
+            currentStyle.negate(command: command)
+        } else {
+            let param = try scanParam()
+            currentStyle.apply(command: command, param: param)
+        }
+    }
+
+    private mutating func scanParam() throws -> Substring? {
+        if (unicodeScalars.suffix(from: currentIndex).starts(with: "<param>".unicodeScalars)) {
```
Comment (alexdeem) [inline Source/Internal/Scanner.swift:95]: possible performance improvement here

File: Source/Internal/Scanner.swift [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            currentIndex = unicodeScalars.index(currentIndex, offsetBy:7)
+            let paramStartIndex = currentIndex
+            guard let closingParamRange = string.suffix(from: currentIndex).range(of:"</param>") else {
```
Comment (alexdeem) [inline Source/Internal/Scanner.swift:98]: possible performance improvement here

**Comment by alexdeem** [comment]
Added some notes for possible performance improvements to be addressed in future.

**Review by ConEliopoulos — APPROVED** [review]
LGTM

**Pull request merged**